civihr_employee_portal/src/View/CustomFieldsMappingFixer.php

```diff
@@ -0,0 +1,164 @@
+        }
+
+        self::processDisplayOptions($displayOption);
+      }
+    }
+  }
+
+  /**
+   * Process all display options for an exported view only when it is related
+   * to a field and the field has a number at the end of its name, which is
+   * the pattern for custom fields from CiviCRM
+   *
+   * @var array
+   *     Display option a exported view structure
+   */
+  private static function processDisplayOptions(&$displayOption) {
+    foreach ($displayOption as &$option) {
+      // only continue if this option is about a field
+      if (!isset($option['field'])) {
+        continue;
+      }
+
+      $optionIdParts = explode('_', $option['field']);
+      $finalIdPart = array_pop($optionIdParts);
+      if (count($optionIdParts) === 0 || !intval($finalIdPart)) {
+        continue;
+      }
+
+      self::processOption($option);
+    }
+  }
+
+  /**
+   * To process a display option for a field in view
+   *
+   * @var array $option
+   *     structure that contains an option for a field in view
+   */
+  private static function processOption(&$option) {
+    // checking if this table is part of the civicrm views integration
+    $tableName = self::getCiviCRMtableName($option['table']);
+    $prefix = $tableName ? self::getPrefix($tableName) : false;
+    if ( $prefix ) {
+      $option['table'] = $tableName;
+      $columns = self::getColumnsFromTable($prefix . $tableName);
+      // if the table where the current field belongs exists does not have it
+      // then we can search for similar fields in the same table and make replacements
+      if (!isset($columns[$option['field']])) {
+        $fieldNameBase = self::removeLastNumberOfString($option['field']);
+        $pos = array_search($fieldNameBase, $columns);
+        if ($pos !== FALSE) {
+          $option['field'] = array_search($columns[$pos], $columns);
+        }
+      }
+    }
+  }
+
+  /**
+   * Removes the number in end of a string
+   * In this context that number is always separated by an underscore
+   *
+   * @param  string $string
+   *     the string which could have underscores or not
+   *
+   * @return string
+   *     the string without the number at the end,
+   *     if no number found at the end returns the same string
+   */
+  private static function removeLastNumberOfString($string) {
+    $pieces = explode('_', $string);
+    $lastIndex = count($pieces) - 1;
+    return $lastIndex > 0 && intval(array_pop($pieces)) ? implode('_', $pieces) : $string;
+  }
+
+  /**
+   * Return an existing CiviCRM table based on the base name of the table,
+   * where "base name of the table" is the name of the table without the number
+   * at the end
+   *
+   * @param string $tableName
+   *     this is the table name exported by views
+   *
+   * @return string
+   *     the name of the existing table if found or an empty string
+   */
+  private static function getCiviCRMtableName($tableName) {
+    global $databases;
+    static $tableNamesMap = [];
```

Comment:
I don't think we should use static function variables, this seems like a good situation to use just a regular class property and fetch it just once. Does this class need to be static or can we make it non-static?

Reply:
@mickadoo this class is static because it does not need to store different properties in multiple instances (don't access instance data). The code for this class was approved here https://github.com/compucorp/civihr-employee-portal/pull/526 but I have to include it because it was causing conflicts with the base branch.